content/docs/components-and-props.md

```diff
@@ -15,58 +15,58 @@ redirect_from:
 prev: rendering-elements.html
 next: state-and-lifecycle.html
 ---
+कौम्पोनॅन्टस हमें UI को स्वतंत्र, पुन: प्रयोज्य टुकड़ो में विभाजित करने में मदद करता है, तथा
+हर हिस्से को अलग मानता है| यह पेज कौम्पोनॅन्टस का परिचय देता है| आपको [यहाँ पर विस्तृत कौम्पोनॅन्ट API reference](/docs/react-component.html) मिल जाएगा|
 
-Components let you split the UI into independent, reusable pieces, and think about each piece in isolation. This page provides an introduction to the idea of components. You can find a [detailed component API reference here](/docs/react-component.html).
+सैद्धांतिक रूप से, कौम्पोनॅन्टस जावास्क्रिप्ट फंक्शन्स जैसा है| वे अर्बिट्ररी इनपुट्स स्वीकार करते है (जिसे "props" बुलाया जाता है ) और React एलिमेंट्स रीटर्न करते हैं जो वर्णन करती है की स्क्रीन पर क्या आना चाहिए|
 
-Conceptually, components are like JavaScript functions. They accept arbitrary inputs (called "props") and return React elements describing what should appear on the screen.
+## फंक्शन्स तथा क्लास कौम्पोनॅन्टस {#function-and-class-components}
 
-## Function and Class Components {#function-and-class-components}
-
-The simplest way to define a component is to write a JavaScript function:
+सबसे सरल तरीका एक कौम्पोनॅन्ट को परिभाषित करने का यह है की एक जावास्क्रिप्ट फंक्शन्स लिखा जाए:
```

Comment:
could be simplified as कौम्पोनॅन्ट को परिभाषित करने कासबसे सरल तरीका एक जावास्क्रिप्ट फंक्शन् है